tests/lookup/tests.py

```diff
@@ -1540,9 +1540,26 @@ def test_multivalued_join_reuse(self):
             Season.objects.get(Exact(F("games__home"), "NY") & Q(games__away="Boston")),
             self.s1,
         )
+        with self.assertNumQueries(1) as ctx:
+            self.assertEqual(
+                Season.objects.get(
+                    Exact(F("games__home"), "NY") & Exact(F("games__away"), "Boston")
+                ),
+                self.s1,
+            )
+        self.assertNotIn("LEFT OUTER", ctx.captured_queries[0]["sql"])
+
+    def test_multivalued_isnull(self):
         self.assertEqual(
-            Season.objects.get(
-                Exact(F("games__home"), "NY") & Exact(F("games__away"), "Boston")
+            Season.objects.get(IsNull(F("games"), True)),
+            self.s2,
+        )
+
+    def test_multivalued_coalesce(self):
+        product = Product.objects.create(qty_target=42)
+        self.assertEqual(
+            Product.objects.get(
+                Exact(Coalesce("stock__qty_available", "qty_target"), 42)
```

Comment:
The exact use case brought up by Anssi a few year ago https://gist.github.com/akaariai/a4c1acbfedac0f2cbf3e